CRM/Financeextras/Hook/BuildForm/AdditionalPaymentButton.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+use CRM_Financeextras_ExtensionUtil as E;
+
+/**
+ * Implement Refund Button if supported Refund by Payment Processor
+ */
+class CRM_Financeextras_Hook_BuildForm__AdditionalPaymentButton {
+
+  /**
+   * @var string
+   *   Path where template with javascript stored for adding new Refund Button.
+   */
+  private $templatePath;
+
+  /**
+   * @var int
+   *   Contribution id for adding submit refund button.
+   */
+  private $contributionID;
+
+  /**
+   * @var \CRM_Contribute_Form_AdditionalPayment
+   *   Form object that is being altered.
+   */
+  private $form;
+
+  /**
+   * @var string
+   *   form name
+   */
+  private $formName;
+
+  /**
+   * CRM_Financeextras_Hook_BuildForm__AdditionalPaymentButton constructor.
+   *
+   * @param \CRM_Contribute_Form_AdditionalPayment $form
+   * @param string $formName
+   */
+  public function __construct(&$form, $formName) {
+    $this->form = $form;
+    $this->formName = $formName;
+    $this->templatePath = E::path() . '/templates/CRM/AdditionalPayment/Form/Payment';
+  }
+
+  public function buildForm(): void {
+    if (!$this->validate()) {
+      return;
+    }
+
+    $this->contributionID = $this->form->getVar('_id');
+
+    $this->addRefundButton();
+  }
+
+  private function validate(): bool {
+    return $this->formName === 'CRM_Contribute_Form_AdditionalPayment';
+  }
+
+  /**
+   * Added template to add the button for submit refund using form-bottom region
+   */
+  private function addRefundButton() {
+    $paymentRefund = new \Civi\Financeextras\Payment\Refund($this->contributionID);
+
+    if (!$paymentRefund->contactHasRefundPermission()) {
```

Comment:
Shouldn't refund eligibility checks stay in side validate method? The method name, `addRefundButton`, should just add a refund button.  

 I think the name `validate` is vague  for the form, since we don't really validate the submission of the form.